component/src/main/java/org/wso2/carbon/identity/authenticator/linkedIn/LinkedInAuthenticator.java

```diff
@@ -74,7 +75,32 @@ public boolean canHandle(HttpServletRequest request) {
         }
         return request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE) != null
                 && request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_STATE) != null
-                && (getLoginType(request));
+                && (getLoginType(request))
+                || request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_ERROR) != null;
+    }
+
+    /**
+     * Handle error response when click on cancel without providing credentials.
+     *
+     * @param request httpServletRequest
+     * @throws InvalidCredentialsException
+     */
+    private void handleErrorResponse(HttpServletRequest request) throws InvalidCredentialsException {
+        if (request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_ERROR) != null) {
+            StringBuilder errorMessage = new StringBuilder();
+            String error = request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_ERROR);
+            String errorDescription = request.getParameter
+                    (LinkedInAuthenticatorConstants.OAUTH2_PARAM_ERROR_DESCRIPTION);
+            String state = request.getParameter(LinkedInAuthenticatorConstants.OAUTH2_PARAM_STATE);
+            errorMessage.append("error: ").append(error)
```

Comment:
Can we make use of constants in the error message as well

Reply:
Fixed in a7af94b5984b94030bc6619fb2306453dbb58597